pkg/crc/preflight/preflight_checks_common.go

```diff
@@ -34,27 +34,31 @@ func fixBundleCached() (bool, error) {
 		if err != nil {
 			return false, err
 		}
-
-		bundleDir := filepath.Base(constants.DefaultBundlePath)
-		err = os.MkdirAll(bundleDir, 0600)
-		if err != nil && !os.IsExist(err) {
-			return false, fmt.Errorf("Cannot create directory %s", bundleDir)
+		available := extractor.AvalibleData()
+		if len(available) < 1 {
+			return false, fmt.Errorf("Invalid bundle data")
 		}
-		f, err := os.Create(constants.DefaultBundlePath)
+		reader, err := extractor.GetReader(available[0])
 		if err != nil {
 			return false, err
 		}
-		available := extractor.AvalibleData()
-		data, err := extractor.ByteArray(available[0])
+		defer reader.Close()
+
+		bundleDir := filepath.Dir(constants.DefaultBundlePath)
+		err = os.MkdirAll(bundleDir, 0700)
+		if err != nil && !os.IsExist(err) {
+			return false, fmt.Errorf("Cannot create directory %s", bundleDir)
+		}
```

Comment:
This would create a directory with the same name as the bundle? Although i don't see any dir created while running this :-/

Reply:
The bundle path is going to be `~/.crc/cache/crc_libvirt_4.2.0.crcbundle`, so `filepath.Dir()` will be `~/.crc/cache/`. Without this, `crc setup` will fail when you run it after `rm -rf ~/.crc/cache`